lox/main.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) assignUser
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fmt/core.h>
+#include <fmt/format.h>
+
+#include <fstream>
+#include <lyra/arg.hpp>
+#include <lyra/help.hpp>
+#include <lyra/lyra.hpp>
+#include <numeric>
+#include <optional>
+#include <tl/expected.hpp>
+#include <tl/optional.hpp>
+#include <variant>
+#include <vector>
```

Comment:
It's pretty common to separate your includes based on:
1. provided by your language / standard
2. provided by your system / platform
3. provided by your third-party dependencies
4. provided by you

The order can be defined by you, but for files with lots of includes, you may find it beneficial to split/organize.

```suggestion
#include <fstream>
#include <optional>
#include <vector>

#include <fmt/core.h>
#include <fmt/format.h>
#include <lyra/arg.hpp>
#include <lyra/help.hpp>
#include <lyra/lyra.hpp>
#include <tl/expected.hpp>
#include <tl/optional.hpp>
```

Reply:
Yeah that makes sense, this is clang today's order. It's probably possible to configure that?

Reply:
Yes, clang-format can help with some ordering

Reply:
It can supposedly also regroup them but that seems a bit excessive (and I didn't get it to work xD).